Katello/katello
# Fixes #36250 - Alert user about past and current filters in the UI

#### What are the changes introduced in this pull request?

Adds:
- A flag to each CV version that had filters applied
- A warning to the CV publish modal that tells users if filters are applied
- A link that opens up the CV filters page in a new link

#### Considerations taken when implementing this change?
This was built on top of @sjha4 's  needs_publish update.

#### What are the testing steps for this pull request?
1) Make filters and check that the CV review modal shows the warning correctly.
2) Remove the filters and check that the warning goes away.
3) Publish some CVVs with filters and ensure the filter icon appears.
4) Check that the filters link works.
5) Check that the "needs publish" icon works well with the filters applied icon.
